### PR TITLE
LADX: Remove duplicate Magnifying Lens item

### PR DESCRIPTION
**Pull request by hatkirby**
## What is this fixing or adding?
Removed the duplicate item (Magnifying Lens), and made the real one a filler item.

## How was this tested?
pytest and a test generation + running the game to make sure it didn't crash.

### CODE REVIEW

**Comment by Seatori** [comment]
I'm not familiar with LADX, but I decided to look this up on the wiki and found it listed there as the "Magnifying Lens", including a screenshot that shows it being called such in-game:
Magnifying_Lens_Obtain
The wiki also mentions that it *is* used in the trading quest, though maybe it's an optional requirement/removed in the mod? Can't say for sure on that as I haven't played it.

**Comment by hatkirby** [comment]
Yeah, someone else asked me this offline, so to clarify: The Magnifying Glass is never required in the randomizer currently. In the base game, you use it to find the path through the Wind Fish's Egg, and to see the invisible Goriya who gives you the Boomerang. In AP, the Magnifying Glass is not needed in order to logically fight the Shadow Nightmare; you are expected to know the path through the Wind Fish's Egg already. The upstream randomizer has a setting for whether or not you need the Magnifying Glass to be able to see the Goriya and get the Boomerang check. This setting is force disabled in the AP randomizer. We may want to change this in the future and allow players to choose to make this check a part of the trade quest. As of right now, though, logic never checks that you have the Magnifying Glass, so it doesn't need to be progression.

The item's name being incorrect is a good point, and it might be a good idea to fix it now.

**Comment by threeandthreee** [comment]
Fyi, upstream has changed the boomerang check to require lens, if we want to go that direction I'd think it should be easy to copy over https://github.com/daid/LADXR/pull/156

**Comment by threeandthreee** [comment]
There's also this one, that makes the enemies in the mermaid scale cave visible https://github.com/daid/LADXR/pull/155

**Comment by threeandthreee** [comment]
As a understand it, the `MAGNIFYING_LENS` lines in  `ladx/LADXR/locations/constants.py` and `ladx/LADXR/locations/items.py` are just for labeling existing data in the rom, I think they should probably just be left as is.

The changes in `ladx/Items.py` should be all that is necessary.


I submitted a PR today bringing in the upstream changes I linked earlier, which add value to the magnifier ( https://github.com/ArchipelagoMW/Archipelago/pull/3829 ). If that is accepted we'd want to keep it as progression but no matter what it makes sense to drop the `MAGNIFYING_LENS` item.

**Review by Exempt-Medic — APPROVED** [review]
Played through a couple games and tested generations. This change matches my understanding of the AP code (checked through a code search and looking through playthroughs). This properly adjusted the classification of the item to no longer be progression.

**Comment by palex00** [comment]
As a person knowledgable with LADX, both upstream and AP, I can confirm that only one item exists and the second item is erronous. I can also confirm that due to the way Archipelago handles the "Boomerang Sales Guy" Check with no requirement other than access to the cave itself, it is not a progression item. 

Upstream has changed this behavior, but AP currently has not.

There is a current PR open that aims to change this:
https://github.com/ArchipelagoMW/Archipelago/pull/3829
but it's in the same limbo as this PR

**Review by palex00 — APPROVED** [review]
As a person knowledgable with LADX, both upstream and AP, I can confirm that only one item exists and the second item is erronous. I can also confirm that due to the way Archipelago handles the "Boomerang Sales Guy" Check with no requirement other than access to the cave itself, it is not a progression item.

Upstream has changed this behavior, but AP currently has not.

There is a current PR open that aims to change this:
https://github.com/ArchipelagoMW/Archipelago/pull/3829
but it's in the same limbo as this PR

**Review by threeandthreee — APPROVED** [review]

**Comment by NewSoupVi** [comment]
I'm currently looking at https://github.com/ArchipelagoMW/Archipelago/pull/3829. I will probably do that one first, and then this one will potentially have to be adjusted, if I understand things correctly.

File: worlds/ladx/Items.py [hunk 1]
```diff
@@ -295,7 +293,7 @@ class ItemName:
     TradeItemData(ItemName.TRADING_ITEM_FISHING_HOOK, "TRADING_ITEM_FISHING_HOOK", trade_item_prog, "Grandma (Animal Village)"),
     TradeItemData(ItemName.TRADING_ITEM_NECKLACE, "TRADING_ITEM_NECKLACE", trade_item_prog, "Fisher (Martha's Bay)"),
     TradeItemData(ItemName.TRADING_ITEM_SCALE, "TRADING_ITEM_SCALE", trade_item_prog, "Mermaid (Martha's Bay)"),
-    TradeItemData(ItemName.TRADING_ITEM_MAGNIFYING_GLASS, "TRADING_ITEM_MAGNIFYING_GLASS", trade_item_prog, "Mermaid Statue (Martha's Bay)")
+    TradeItemData(ItemName.TRADING_ITEM_MAGNIFYING_GLASS, "TRADING_ITEM_MAGNIFYING_GLASS", ItemClassification.filler, "Mermaid Statue (Martha's Bay)")
```
Comment (threeandthreee) [inline worlds/ladx/Items.py:296]: now that https://github.com/ArchipelagoMW/Archipelago/pull/3829 is merged:
```suggestion
    TradeItemData(ItemName.TRADING_ITEM_MAGNIFYING_GLASS, "TRADING_ITEM_MAGNIFYING_GLASS", trade_item_prog, "Mermaid Statue (Martha's Bay)")
```

**Review by threeandthreee — CHANGES_REQUESTED** [review]

**Review by threeandthreee — APPROVED** [review]

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**